### ISSUE TITLE
Option to never filter Self

### ISSUE THREAD

**Issue by blitzduck** [jackriccomini/spamfilter-plugin-runelite#17]
Sometimes I will send a message that gets filtered (to myself only) but other players will see it.

There should certainly be an option to NEVER filter your own messages (by default, tbh...)

There have been numerous times I sent the same message multiple times because it didn't show up on my screen. other users just say, "yeah... you said that..." embarassing :P

**Comment by jackriccomini** [comment]
There is an open PR (thank you TreeOtter!) to be merged fairly soon which fixes this. Once it has been merged and I've made a PR to the RuneLite plugin-hub for their team to review, I will comment a link to that PR here so you can follow updates.

**Comment by jackriccomini** [comment]
This is being reviewed by the plugin hub [here](https://github.com/runelite/plugin-hub/pull/5452)

**Comment by SkrumpKing** [comment]
Looking forward to this update! The spam filter has been working wonders for the GE experience and this has been the only issue I've seen so far.

**Comment by loganthepage** [comment]
Looks like this has been stalled for a while, any hope of it working?

**Comment by cucumbercoast** [comment]
> Looks like this has been stalled for a while, any hope of it working?

The fix was rejected because of [deprecated APIs](https://github.com/runelite/plugin-hub/pull/5452#issuecomment-1913867130). I'd also really appreciate if there was progress on this. It looks like it was last talked about [4 months ago](https://github.com/jackriccomini/spamfilter-plugin-runelite/issues/12#issuecomment-2040673815) though.

**Comment by jackriccomini** [comment]
Yep, it's been stalled for a while because of the deprecated APIs mentioned above. I haven't looked at this repository in a while because I haven't been playing recently. Don't think that your comments go into the void - I haven't been playing the game or developing the plugin but I do see the email notifications GitHub sends when someone comments on an issue in this repository and I make a mental note that I should come back to it when I next feel like writing some code outside of work.

Until then, I'd be happy to review and accept patches if anyone else wanted to tackle the deprecated API problem that blocks the fix for this issue from being accepted into the plugin hub.
